From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: amaanq <git@amaanq.com>
Date: Mon, 10 Feb 2026 00:00:00 +0000
Subject: [PATCH] enable RANDSTRUCT_FULL by default

Enable full structure layout randomization by default. This is a major
exploit mitigation that randomizes the layout of sensitive kernel
structures at compile time, requiring an additional information exposure
vulnerability for exploits targeting these structure types.

Requires Clang >= 16 or GCC with randstruct plugin (bunkernel uses
Clang).

Signed-off-by: amaanq <git@amaanq.com>
---
 security/Kconfig.hardening | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index 2cfc7dd457d6..3a06082487a1 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -286,7 +286,7 @@ config CC_HAS_RANDSTRUCT
 choice
 	prompt "Randomize layout of sensitive kernel structures"
 	default RANDSTRUCT_FULL if COMPILE_TEST && (GCC_PLUGINS || CC_HAS_RANDSTRUCT)
-	default RANDSTRUCT_NONE
+	default RANDSTRUCT_FULL
 	help
 	  If you enable this, the layouts of structures that are entirely
 	  function pointers (and have not been manually annotated with
--
2.48.1

